class Solution {
    public int[] findErrorNums(int[] nums) {
        int i = 0;
        while (i < nums.length) {
            int correct = nums[i] - 1;
            if (nums[i] != nums[correct]) {
                Swap(nums, i, correct);
            } else {
                i++;
            }
        }

            for (int j = 0; j<nums.length; j++){
                if (nums[j] != j+1) {
                    return new int[]{nums[j], j+1};
                }
            }
            return new int[] {-1, -1};
        }
        void Swap(int[] arr, int i, int correct) {
        int temp = arr[i];
        arr[i] = arr[correct];
        arr[correct] = temp;
    }
} 